File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"

bool is_armstrong_number(int candidate)
{
    int digits = 0;
    int num = candidate;
    int tens[] = {1, 10, 100, 1000, 10000, 100000, 1000000, 10000000, 100000000, 1000000000};
    if (num == 0)
        digits++;
    while (num > 0)
    {
        num /= 10;
        digits++;
    }
    int sum = 0;
    for (int i=0; i<digits; i++)
    {
        int digit = candidate / tens[digits - i - 1] % 10;
        int sumDigit = digit;
        for (int j=1; j<digits; j++)
            sumDigit *= digit;
        sum += sumDigit;
    }
    return (candidate == sum);
}

File: solutions/python/armstrong-numbers/1/armstrong_numbers.py
"""
A function for finding Armstrong numbers.

An Armstrong number is a number that is the sum of its own digits each raised to the power of the number of digits.
"""

def is_armstrong_number(number):
    n = number
    num_digits = 0
    while n >= 1:
        n /= 10
        num_digits += 1
    if number == 0:
        num_digits = 1
    digits = [int(d) for d in str(number)]
    return number == sum(map(lambda x: x ** num_digits, digits))